lib/sgtn-client/util/locale-util.rb

```diff
@@ -11,14 +11,17 @@ module SgtnClient
 
     class LocaleUtil
         def self.fallback(locale)
+            if locale.nil?
+                return nil
+            end
             found = SgtnClient::DEFAULT_LOCALES.select {|e| e == locale}
             if !found.empty?
                 return found[0]
             end
             if SgtnClient::MAP_LOCALES.key?(locale)
                 return SgtnClient::MAP_LOCALES[locale]
             end
-            parts = locale.split("-")
+            parts = locale.to_s.split("-")
```

Comment:
This is for a different issue: https://github.com/vmware/singleton/issues/1479
Please move to a different PR.